### PR TITLE
more logging, more items

### PR DESCRIPTION
**Pull request by dennistruemper**
Hi, I wanted to add a feature to add a pattern to only delete test, beta, you name it branches. 
This is what I came up with by now. I will test it on a project of mine and come back.

### CODE REVIEW

File: README.md
Comment (thadguidry) [inline README.md:48]:
```suggestion
Specifies a pattern to match. If not specified then every release will be targeted. If specified then every release containing the pattern will be targeted. Use this option for example to remove old beta releases.
```

File: action.yml
Comment (thadguidry) [inline action.yml:20]:
```suggestion
    description: part of the tag name. Example, if you want to delete 0.0.1-beta and 0.0.2-beta but not 0.0.1 then set this to just "beta". If not set then it will target all releases.
```

**Review by thadguidry — CHANGES_REQUESTED** [review]

**Comment by dennistruemper** [comment]
@thadguidry I am Sorry, I can't get this to work...
Bildschirmfoto 2020-12-04 um 21 56 29

These were my settigs:
`with:
          keep_latest: 10
          delete_tags: true
          delete_tag_pattern: test
        env:
          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}`

Do you have a clue? I don't get, the clients receives theese Ids and right after they can't be found? I tried to change the access Token, didn't help.

So if you don't want to invest time into this, just close it, I will serach for an other solution.

**Comment by thadguidry** [comment]
@dennistruemper You have the right idea overall and it's definitely doable and I think quite valuable for the entire community.  I think you just need to continue working through the problem with the GitHub API and the results that come back and what's available with the data returned :-) 
Most importantly, fully understand it from GitHub API perspective...reading through this entire page and scrolling slowly through it and reading all highlighted notes, examples, etc:
https://developer.github.com/v3/repos/releases/
Don't give up, take the weekend or weeks!

**Comment by dennistruemper** [comment]
@thadguidry wow, what a stupid error. Wrong variable name in log statement, no wrong release Id. 
It worked. I removed more than 100 old test releases

**Comment by dennistruemper** [comment]
@thadguidry for me it is late by now. I am waiting for the next release of your project to remove my fork and switch my project back to your GH Action again :-)

Thanks for pushing me

Have a nice Weekend

**Comment by thadguidry** [comment]
@dennistruemper it's not mine... it's the entire communty's.  But I'm sure @dev-drprasad will review this and get back to you over the weekend. Enjoy!

**Comment by dev-drprasad** [comment]
> @thadguidry wow, what a stupid error. Wrong variable name in log statement, no wrong release Id.
> It worked. I removed more than 100 old test releases

sometimes I hate `try...catch`

File: index.js
Comment (dev-drprasad) [inline index.js:63]: **Nit:**

I just think `console.log("all releases will be targeted");` is redundant. SInce `delete_tag_pattern` is optional, we dont need tell we're targeting all releases. (In my opinion)

I am proposing
```suggestion
let deletePattern = process.env.INPUT_DELETE_TAG_PATTERN || "";
if (deletePattern) {
  console.log(`releases containing ${deletePattern} will be targeted`);
}
```

File: index.js
Comment (dev-drprasad) [inline index.js:80]: Any reason you mentioned `per_page=100` ?

File: index.js
Comment (dev-drprasad) [inline index.js:88]: **Nit:**

May be we can move filter predicate inside next line. Why do extra work if there is compelling reason.

something like:

```suggestion
    // filter for delete_pattern
    const activeMatchedReleases = data.filter(({draft}) => !draft && tag_name.indexOf(deletePattern) !== -1)

```

**Comment by dev-drprasad** [comment]
@dennistruemper Thanks for PR. Its really useful feature that people can use. Give me some time to test it for myself also.

@thadguidry Thanks for helping people out

**Comment by dev-drprasad** [comment]
> @thadguidry I am Sorry, I can't get this to work...
> Bildschirmfoto 2020-12-04 um 21 56 29
> 
> These were my settigs:
> `with: keep_latest: 10 delete_tags: true delete_tag_pattern: test env: GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}`
> 
> Do you have a clue? I don't get, the clients receives theese Ids and right after they can't be found? I tried to change the access Token, didn't help.
> 
> So if you don't want to invest time into this, just close it, I will serach for an other solution.

In case you're not aware, You can test locally by running

```
GITHUB_TOKEN=<token> GITHUB_REPOSITORY=<repo/name> INPUT_KEEP_LATEST=0 node index.js
```

File: index.js
Comment (dennistruemper) [reply index.js:80]: default is 30, maximum is 100. I didn't want to add full pagination support too. My project hat ~170 Releases and i Wanted to keep al the production Releases (around 25) so there were only 5 spots per execution for prereleases to be deleted.

My strategy now is to keep 10 of the releases containing 'test' and 70 releases in total. That way I can see the Releases of the last ~year (60 weeks, I try to release one a week)

For this Strategy to work with "per_page=100" I can have 20 releases to be deleted between clean up runs.

**Comment by dennistruemper** [comment]
> > @thadguidry I am Sorry, I can't get this to work...
> > Bildschirmfoto 2020-12-04 um 21 56 29
> > These were my settigs:
> > `with: keep_latest: 10 delete_tags: true delete_tag_pattern: test env: GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}`
> > Do you have a clue? I don't get, the clients receives theese Ids and right after they can't be found? I tried to change the access Token, didn't help.
> > So if you don't want to invest time into this, just close it, I will serach for an other solution.
> 
> In case you're not aware, You can test locally by running
> 
> ```
> GITHUB_TOKEN=<token> GITHUB_REPOSITORY=<repo/name> INPUT_KEEP_LATEST=0 node index.js
> ```

thats how I found the stupid try catch bug in the end. But I also commented out the delete fetches. Some sort of dry run would help with development

**Comment by dennistruemper** [comment]
> @dennistruemper Thanks for PR. Its really useful feature that people can use. Give me some time to test it for myself also.
> 
> @thadguidry Thanks for helping people out

Hi @dev-drprasad , 
thanks for the quick review. I added your suggestions, good points :-) .
Take your time with testing, I am using my fork until it is merged.

This looked like a quick win, so i thought I'll try it.

File: index.js
Comment (dev-drprasad) [inline index.js:81]: @dennistruemper  I see you merged my suggestion. But it is missing `tag_name` inside filter predicate parameters. Sorry!

Comment (dev-drprasad) [reply index.js:81]: Also `activeMatchedReleases` not used in code. Please make that change also

Comment (dennistruemper) [reply index.js]: @dev-drprasad I fixed that now, sorry did not see that and didn't check. This time I ran it local, it works now for me.

We will get this working ;-)

**Comment by dev-drprasad** [comment]
@dennistruemper  🎉 I merged your PR and released [v0.2.0](https://github.com/dev-drprasad/delete-older-releases/releases/tag/v0.2.0) with new changes.

Thanks for PR!

**Pull request merged**
